### PR TITLE
Vet clinic database: add "join table" for visits

### PR DESCRIPTION
**Pull request by ShiroYaksha90**
# Hi 👋 in this pull request I:

- Create a table named vets with the following columns:

  - id: integer (set it as autoincremented PRIMARY KEY)
  - name: string
  - age: integer
  - date_of_graduation: date

- Create a "join table" called specializations to handle the relationship between the tables' species and vets.
- Create a "join table" called visits to handle the relationship between the tables' animals and vets.
- Insert the following data for vets:
  - Vet William Tatcher is 45 years old and graduated Apr 23rd, 2000.
  - Vet Maisy Smith is 26 years old and graduated Jan 17th, 2019.
  - Vet Stephanie Mendez is 64 years old and graduated May 4th, 1981.
  - Vet Jack Harkness is 38 years old and graduated Jun 8th, 2008.

- Insert the following data for specialties:
  - Vet William Tatcher is specialized in Pokemon.
  - Vet Stephanie Mendez is specialized in Digimon and Pokemon.
  - Vet Jack Harkness is specialized in Digimon.

- Insert the following data for visits:
  - Agumon visited William Tatcher on May 24th, 2020.
  - Agumon visited Stephanie Mendez on Jul 22th, 2020.
  - Gabumon visited Jack Harkness on Feb 2nd, 2021.
  - Pikachu visited Maisy Smith on Jan 5th, 2020.
  - Pikachu visited Maisy Smith on Mar 8th, 2020.
  - Pikachu visited Maisy Smith on May 14th, 2020.
  - Devimon visited Stephanie Mendez on May 4th, 2021.
  - Charmander visited Jack Harkness on Feb 24th, 2021.
  - Plantmon visited Maisy Smith on Dec 21st, 2019.
  - Plantmon visited William Tatcher on Aug 10th, 2020.
  - Plantmon visited Maisy Smith on Apr 7th, 2021.
  - Squirtle visited Stephanie Mendez on Sep 29th, 2019.
  - Angemon visited Jack Harkness on Oct 3rd, 2020.
  - Angemon visited Jack Harkness on Nov 4th, 2020.
  - Boarmon visited Maisy Smith on Jan 24th, 2019.
  - Boarmon visited Maisy Smith on May 15th, 2019.
  - Boarmon visited Maisy Smith on Feb 27th, 2020.
  - Boarmon visited Maisy Smith on Aug 3rd, 2020.
  - Blossom visited Stephanie Mendez on May 24th, 2020.
  - Blossom visited William Tatcher on Jan 11th, 2021.

- Write queries to answer the following:
  - Who was the last animal seen by William Tatcher? William Tatcher

  - How many different animals did Stephanie Mendez see? Checked

  - List all vets and their specialties, including vets with no specialties. List all vets and their specialties
 
   - List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020. List all animals that visited Stephanie Mendez

   - What animal has the most visits to vets? most visits

   - Who was Maisy Smith's first visit? first visit

   - Details for most recent visit: animal information, vet information, and date of visit. Details

   - How many visits were with a vet that did not specialize in that animal's species? How many visits were with a vet that did not specialize

   - What specialty should Maisy Smith consider getting? Look for the species she gets the most. What specialty should Maisy Smith

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -100,4 +100,80 @@ WHERE owners.full_name = 'Dean Winchester' AND escape_attempts = 0;
...
+JOIN vets ON vets.id = visits.vet_id
+ORDER BY visits.date_of_visit DESC LIMIT 1;
+
+---How many visits were with a vet that did not specialize in that animal's species?
+
+SELECT COUNT(visits.animal_id) AS unspecialized_visits_count FROM visits
+JOIN vets ON visits.vet_id = vets.id
+JOIN animals ON visits.animal_id = animals.id
+JOIN specializations ON specializations.species_id = vets.id
+WHERE specializations.species_id != animals.species_id;
```
Comment (mikethreels) [inline queries.sql:168]: - [x] this query returns a number of 8 visits according to your screenshots however, looking at the data in the tables maisy smith has had 9 visits.

File: data.sql [hunk 2]
```diff
@@ -54,4 +54,85 @@ UPDATE animals SET owner_id = CASE
...
+    '2019-05-15'),
+  ((select id from animals where name like 'Boarmon'),
+    (select id from vets where name like 'Maisy Smith'),
+    '2020-02-27'),
+  ((select id from animals where name like 'Boarmon'),
+    (select id from vets where name like 'Maisy Smith'),
+    '2020-08-03'),
+  ((select id from animals where name like 'Boarmon'),
+    (select id from vets where name like 'Stephanie Mendez'),
+    '2020-05-24'),
```
Comment (mikethreels) [inline data.sql]: - [x] according to the provided data this should be Blossom instead of Boarmon

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Basim, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- good job creating the tables.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ShiroYaksha90** [comment]
Hi @mikethreels I'll implement the changes immediately

File: queries.sql (hunk 1 above)
Comment (ShiroYaksha90) [reply queries.sql:168]: According to the data I inserted it's the value is 8
Visits

File: data.sql (hunk 2 above)
Comment (ShiroYaksha90) [reply data.sql]: Made the required adjustment 👍

**Review by andersonlebon — APPROVED** [review]
## Hi @ShiroYaksha90 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

nodding-yes

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ShiroYaksha90** [comment]
@andersonlebon Thanks 🙏

**Pull request merged**